src/app/features/activity-list/components/transaction-list/transaction-list.tsx

```diff
@@ -10,22 +15,55 @@ interface TransactionListProps {
   bitcoinTxs: TransactionListBitcoinTx[];
   stacksTxs: TransactionListStacksTx[];
 }
+
 export function TransactionList({ bitcoinTxs, stacksTxs }: TransactionListProps) {
+  const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const currentBitcoinAddress = nativeSegwitSigner.address;
+
+  const [visibleTxsNum, setVisibleTxsNum] = useState(10);
+  const { ref: intersectionSentinel, inView } = useInView();
+
+  useEffect(() => {
+    setVisibleTxsNum(visibleTxsNum + 10);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inView]);
+
+  useEffect(() => {
+    setVisibleTxsNum(10);
+  }, [currentBitcoinAddress]);
+
   const txsGroupedByDate = useMemo(
     () =>
       bitcoinTxs.length || stacksTxs.length ? createTxDateFormatList(bitcoinTxs, stacksTxs) : [],
     [bitcoinTxs, stacksTxs]
   );
 
+  const groupedByDateTxsLength = useMemo(() => {
+    return txsGroupedByDate.reduce((acc: Record<string, number>, item, index) => {
+      acc[index] = item.txs.length + (acc[index - 1] || 0);
+      return acc;
+    }, {});
+  }, [txsGroupedByDate]);
+
   return (
     <TransactionListLayout>
-      {txsGroupedByDate.map(({ date, displayDate, txs }) => (
-        <TransactionsByDateLayout date={date} displayDate={displayDate} key={date}>
-          {txs.map(tx => (
-            <TransactionListItem key={getTransactionId(tx)} tx={tx} />
-          ))}
-        </TransactionsByDateLayout>
-      ))}
+      {txsGroupedByDate.map(({ date, displayDate, txs }, dateIndex) => {
+        if (groupedByDateTxsLength[dateIndex] > visibleTxsNum) {
+          return null;
+        }
+
+        return (
+          <TransactionsByDateLayout date={date} displayDate={displayDate} key={date}>
+            {txs.map((tx, txIndex) => {
+              if ((groupedByDateTxsLength[dateIndex - 1] || 0) + txIndex > visibleTxsNum)
```

Comment:
maybe make a function here so this is more readable?

Reply:
refactored a bit, to make code more clear